src/window/mod.rs

```diff
@@ -265,12 +268,38 @@ pub fn create_window(
     let event_loop = EventLoop::new();
 
     let cmd_line_settings = SETTINGS.get::<CmdLineSettings>();
+    let frame_decoration = cmd_line_settings.frame;
+
+    #[cfg(not(target_os = "macos"))]
     let winit_window_builder = window::WindowBuilder::new()
         .with_title("Neovide")
         .with_window_icon(Some(icon))
         .with_maximized(cmd_line_settings.maximized)
         .with_transparent(true)
-        .with_decorations(!cmd_line_settings.frameless);
+        // There is only two options for windows & linux, no need to match more options.
+        .with_decorations(frame_decoration == Frame::Full);
+
+    #[cfg(target_os = "macos")]
+    let winit_window_builder = match frame_decoration {
+        Frame::Full => window::WindowBuilder::new()
+            .with_title("Neovide")
+            .with_window_icon(Some(icon))
+            .with_maximized(cmd_line_settings.maximized),
+        Frame::None => window::WindowBuilder::new()
+            .with_title("Neovide")
+            .with_window_icon(Some(icon))
+            .with_maximized(cmd_line_settings.maximized)
+            .with_decorations(false),
+        Frame::Buttonless => window::WindowBuilder::new()
+            .with_window_icon(Some(icon))
+            .with_maximized(cmd_line_settings.maximized)
+            // This option removes the shadow & makes the margin color black(?)
+            //.with_has_shadow(false)
+            .with_title_hidden(true)
+            .with_titlebar_buttons_hidden(true)
+            .with_titlebar_transparent(true)
+            .with_fullsize_content_view(true),
+    };
```

Comment:
The match arm type is incompatible, we need to do something like this in the first arm:
``` 
Frame::Full => winit_window_builder,
```
and clone the icon value in first window builder. What do you think?

Reply:
I've updated the gist, because I missed an important detail.
Now we create the builder once (which is what I meant originally).
Not sure what do you mean about cloning the icon value.

Reply:
Now I think that the first two matches are obsolete and it can be just:
```rust
#[cfg(target_os = "macos")]
let winit_window_builder = if frame_decoration == Frame::Buttonless {
        winit_window_builder
            .with_title_hidden(true)
            .with_titlebar_buttons_hidden(true)
            .with_titlebar_transparent(true)
            .with_fullsize_content_view(true),
        } else {
          winit_window_builder
        };
```
Thoughts?

Reply:
That much better and it really simplify the code.

The icon value was moved first so we can't use it again, but in the updated gist there is no need to clone it.

Hmmm I think you are right about the first two being obsolete, but we can add an extra option like Alacritty did

```rust
let winit_window_builder = match frame_decoration {
    Frame::Full => winit_window_builder,
    Frame::None => winit_window_builder
        .with_decorations(false),
    Frame::Buttonless => winit_window_builder
        // This option removes the shadow & makes the margin color black(?)
        //.with_has_shadow(false)
        .with_title_hidden(true)
        .with_titlebar_buttons_hidden(true)
        .with_titlebar_transparent(true)
        .with_fullsize_content_view(true),
    Frame::Transparent => winit_window_builder
        .with_title_hidden(true)
        .with_titlebar_transparent(true)
        .with_fullsize_content_view(true),
```
I don't know the benefit of the transparent, but we can add it as an option for the user. What do you say?

Reply:
I'm not sure what all these options do (I don't use macos).
I guess we could add it while we're at it.

Reply:
I just updated the code and added .vscode folder to .gitignore

Let me know if more changes are needed.